src/pq/conninfo.cr

```diff
@@ -33,6 +42,9 @@ module PQ
       @user = default_user user
       @port = (port || 5432).to_i
       @sslmode = default_sslmode sslmode
+      @sslcert = ENV["PGSSLCERT"]?
+      @sslkey = ENV["PGSSLKEY"]?
+      @sslrootcert = ENV["PGSSLROOTCERT"]?
```

Comment:
I think this would become the only place where the library is directly looking at environment variables. Would it be possible to make this be an argument, and have people pick how they get this info into the library, be it env vars, or some other means. I realize that these are probably the env vars that libpq uses, but it would be nice to be a little more flexible.

It may also make sense to make a struct that holds these three values, if that makes passing it around easier.

Reply:
Thought that using the predefined PostgreSQL environment variables could a be sensible default because it is compatible with `libpq` https://www.postgresql.org/docs/current/static/libpq-envars.html. However i understand that this should probably also include the other `PG*` environment variables. At the moment i build an url like this:

```crystal
class DBUri < URI
  def initialize(**args)
    self.scheme = "postgres"
    self.user = ENV["PGUSER"]
    self.password = ENV["PGPASSWORD"]?
    self.host = ENV["PGHOST"]?
    self.port = ENV["PGPORT"]?.try &.to_i
    self.path = "/" + ENV["PGDATABASE"]
  end
end
```

db = DB.open DBUri.new
``` 

However i am unsure how to pass in the certificate information because all the `crystal-db` api's seem to use urls, such as the `DB.open` interface http://crystal-lang.github.io/crystal-db/api/0.5.0/DB.html#open%28uri%3AURI%7CString%29-class-method.

Reply:
Used this article for my setup https://www.gab.lc/articles/postgresql_with_ssl/, but i am unsure how i should add these steps to the spec...

Reply:
Now using query parameters to pass in the ssl parameters instead of relying environment variables.